src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -2585,7 +2616,8 @@ def _array_syntax_to_indexed(self, parent, loop_vars):
             for idx, child in enumerate(array.children):
                 if isinstance(child, Range):
                     symbol = _find_or_create_imported_symbol(
-                                array, loop_vars[range_idx])
+                        array, loop_vars[range_idx],
+                        symbol_type=DataSymbol, datatype=DeferredType())
```

Comment:
Can we infer that Range indexes are of Integer Datatype?

Reply:
Mmm, we could but we can't infer the precision so I think we should leave this. (Unless we can have an integer of unknown precision?)

Reply:
After I wrote the comment I thought that even if we know the datatype it expects it may be possible to have some implicit casts to that type, so symbol datatype inference may be very tricky. The inference of the symbol_type is much easier because it just depends on where the symbol it is found. So I agree with your decision to leave it as DeferedType.